Add initial default CONTRIBUTING.md file

brianwarner opened the pull request:
This is the initial commit of a default CONTRIBUTING.md file, which will
replicate across all repositories which do not currently have a local
CONTRIBUTING.md.

This PR should be approved by the TSC prior to being merged.

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add initial default CONTRIBUTING.md file

This is the initial commit of a default CONTRIBUTING.md file, which will
replicate across all repositories which do not currently have a local
CONTRIBUTING.md.

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add instructions on becoming a committer

This commit adds details on the process to become a committer, and clarifies how
new committer positions are added or existing positions removed.

Signed-off-by: Brian Warner <brian@bdwarner.com>

brianwarner committed:
Add current committers with affiliations

Signed-off-by: Brian Warner <brian@bdwarner.com>

caniszczyk commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,75 @@
+| Jiexi Lin         |[jessesleeping](https://github.com/jessesleeping)  | Facebook    |
+
+
+The path to committership starts with becoming a contributor to the project. Over time contributors prove that they have high commitment to the project and they get nominated by an existing committer to become a committer.
```
maybe add that a simple majority vote is needed to add an additional committer

brianwarner replied:
Thanks for pointing this out, @caniszczyk. I attempted to clarify it, could you please have a look at what I just pushed?

brianwarner committed:
Add specificity to committer selection

The Presto Technical Charter provides two paths to committership: either a
simple majority vote by existing committers, or by a 2/3 vote by the TSC.
Likewise, the same thresholds may remove an existing committer.

This patch cleans up the language and clarifies the above.

Signed-off-by: Brian Warner <brian@bdwarner.com>

nezihyigitbasi commented on CONTRIBUTING.md:
```diff
@@ -12,17 +12,34 @@ All Presto development occurs on [GitHub](https://github.com/prestodb), followin
+
+When a committer candidate is nominated by an existing committer lazy consensus is required (no vetoes) from existing committers. The nomination should include a justification of why that person should become a committer (e.g., code/test/doc contributions, other positive impact to the project & community, etc.).
+
+Committer candidates:
```
Now that we specify committer criteria in CONTRIBUTING.md file, do we still want to keep the same/similar info in the [wiki](https://github.com/prestodb/presto/wiki/How-to-become-a-Presto-committer%3F)?

brianwarner replied:
It's a good question.  I have a few more things to add to the agenda for tomorrow's meeting, I can add this as a topic.

caniszczyk replied:
I'd drop the wiki for listing this type of stuff

nezihyigitbasi commented on CONTRIBUTING.md:
```diff
@@ -12,17 +12,34 @@ All Presto development occurs on [GitHub](https://github.com/prestodb), followin
+* Help with reviewing pull requests from the community
+* Contribute to design/technical discussions on Github and Slack
+* Help with user issues on Github, user mailing list, and Slack
+have good judgement around when some code should be merged, or when to ask someone else to make that judgement
```
Missing `*`.

brianwarner replied:
Thanks @nezihyigitbasi.  I'll push a patch with that correction right now.

brianwarner committed:
Fix list formatting

The list of committer attributes was missing a '*'.

Signed-off-by: Brian Warner <brian@bdwarner.com>

rongrong commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,72 @@
+| Rongrong Zhong    |[rongrong](https://github.com/rongrong)            | Facebook    |
+| Shixuan Fan       |[shixuan-fan](https://github.com/shixuan-fan)      | Facebook    |
+| Jiexi Lin         |[jessesleeping](https://github.com/jessesleeping)  | Facebook    |
+
```
We recently just added a new committer:
``` 
| Leiqing Cai  | [caithagoras](https://github.com/caithagoras) | Facebook |
```
Could you please add him in. Thanks!

brianwarner replied:
Thanks @rongrong, will do!  I'll push a patch now for that, and also adding Venki to the TSC.

brianwarner committed:
Add Leiqing Cai as committer

Per vote of the TSC, Leiqing is now a Presto committer.

Signed-off-by: Brian Warner <brian@bdwarner.com>

galemccommons commented:
Merging. Thanks @brianwarner
